src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -48,6 +48,11 @@ internal AssemblyScanner(Assembly assemblyToScan)
         /// </summary>
         public bool ScanAppDomainAssemblies { get; set; } = true;
 
+        /// <summary>
+        /// Determines if the scanner should scan assemblies from the application directory.
+        /// </summary>
+        public bool ScanFileSystemAssemblies { get; set; } = true;
```

Comment:
I think the name is ok, but we could drop `Assemblies` as its part of `AssemblyScanner`:
```suggestion
        public bool ScanFileSystem { get; set; } = true;
```

Reply:
I'm not entirely sure if we do this but we could also inverse the property name to better imply the default?

```suggestion
        public bool DisableFileSystemScanning { get; set; };
```